### PR TITLE
Add docs for jupyter notebooks

### PR DESCRIPTION
**Pull request by misohu**
Closes:  https://github.com/canonical/data-science-stack/issues/107

To run documentation locally 
```
cd docs
make run 
```

### CODE REVIEW

File: docs/how-to/jupyter-notebook/notebook-logs.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   .. code-block:: bash
+
+       dss logs my-notebook
+
+   Expected output:
+
+   .. code-block:: none
+
+       Logs for my-notebook:
+       [logs]
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook/notebook-logs.rst:43]: Could we show some example logs here as well?

File: docs/how-to/jupyter-notebook/connect-mlflow.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    c.create_experiment("test-experiment")
+
+This example shows how to initialise the MLFlow client, check the tracking URI, and create a new experiment. The `MLFLOW_TRACKING_URI` should already be set in your environment, allowing you to focus on your experiments without manual configuration.
+
+Further Information
+-------------------
+
+For more detailed information on using MLFlow, including advanced configurations and features, refer to the official MLFlow documentation:
+
+- [MLFlow Documentation](https://mlflow.org/docs/latest/index.html)
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook/connect-mlflow.rst:68]: This is not rendering as expected 
image

We'll need to use links similarly to here
https://github.com/canonical/data-science-stack/blob/main/docs/index.rst?plain=1#L62
https://github.com/canonical/data-science-stack/blob/main/docs/reuse/links.txt#L8

File: docs/how-to/jupyter-notebook/delete-notebook.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+Delete a Notebook
+=================
+
+This guide provides instructions on how to remove a Jupyter Notebook from the Data Science Stack (DSS) environment.
+
+Overview
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook/delete-notebook.rst:6]: I would like us to have a note here to tell users that some of the data (stored on the PVCs) will not get lost.

Will depend on the next step we'll agree on https://github.com/canonical/data-science-stack/issues/102. Keeping this comment for visibility and let's sync in parallel on the other issue on what our plan will be for the shared PVC for `v0.1` to then decide what to document here

File: docs/how-to/jupyter-notebook/connect-mlflow.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+--------
+
+MLFlow is a platform for managing the end-to-end machine learning life cycle. It includes tracking experiments, packaging code into reproducible runs, and sharing and deploying models. DSS environments are pre-configured to interact with an MLFlow server through the `MLFLOW_TRACKING_URI` environment variable set in each notebook.
+
+Prerequisites
+-------------
+
+Before you begin, ensure the following:
+
+- You have a Jupyter Notebook running in the DSS environment.
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook/connect-mlflow.rst:16]: can we swap the word `running` with `active` here, so that we are consistent with the terminology of the `dss list` command?

File: docs/how-to/jupyter-notebook/list-status-notebooks.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+- **Active**: The notebook is running and accessible. The URL under the "URL" column will be displayed, allowing you to access the notebook directly.
+
+- **Stopped**: The notebook is not running. You can start it using the `dss start` command.
+
+- **Stopping**: The notebook is in the process of stopping. It is advisable to wait until the process completes, transitioning to "Stopped".
+
+- **Starting**: The notebook is starting up. This state indicates that the notebook is initialising and will soon be active.
+
+- **Downloading**: The notebook is downloading necessary data or software. This is usually a transient state before it becomes "Active".
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook/list-status-notebooks.rst:35]: Can we rephrase this to something like:
` The notebook is downloading the specified OCI Image. This is usually a .....`

To make it clear what's getting downloaded at this step?

**Review by kimwnasptd — COMMENTED** [review]
Really nice work @misohu! Left some small comments

File: docs/how-to/jupyter-notebook/delete-notebook.rst (hunk 3 above)
Comment (misohu) [reply docs/how-to/jupyter-notebook/delete-notebook.rst:6]: Added a note based on the agreed resolution from v.1. Should we keep this comment ?

Comment (kimwnasptd) [reply docs/how-to/jupyter-notebook/delete-notebook.rst:6]: Looks good!

**Review by kimwnasptd — APPROVED** [review]
